pkg/apis/core/validation/shoot.go

```diff
@@ -848,11 +854,36 @@ func fieldNilOrEmptyString(field *string) bool {
 	return field == nil || len(*field) == 0
 }
 
-func validateNetworking(networking core.Networking, fldPath *field.Path) field.ErrorList {
+func validateNetworking(networking *core.Networking, workerless bool, fldPath *field.Path) field.ErrorList {
 	allErrs := field.ErrorList{}
 
-	if len(networking.Type) == 0 {
-		allErrs = append(allErrs, field.Required(fldPath.Child("type"), "networking type must be provided"))
+	if workerless {
+		// Nothing to be validated here, exit
+		if networking == nil {
+			return allErrs
+		}
```

Comment:
We need a service CIDR

Reply:
There can be the case user doesn't want service CIDR, In that case, we don't need service CIDR. 
For the case service CIDR is not added by gardener, kubernetes default service cidr is considered for any service, which is `10.0.0.0/24`.

Reply:
I think it's safer to not rely on implicit defaults of Kubernetes (which can change) and rather let people explicitly define this. If they don't want to provide it - fine, just like for regular `Shoot`s, we can default it based on the `Seed` spec. However, I think we should always explicitly set this in the `Shoot`.

Reply:
There will be very few cases when the service can be used. What if users don't want to create a service, in that case making the service mandatory doesn't help.

Reply:
> we can default it based on the Seed spec

We still do this for workerless Shoots.

> However, I think we should always explicitly set this in the Shoot.

Currently serviceCIDR is optional for even shoot with workers. There are no fields inside Networking which are not optional for workerless Shoots. What would the user put in the networking field in the Shoot yaml, if they want it to be defaulted by Gardener?

One thing I can think of is, currently for workerless, if the service CIDR is neither present in the Shoot nor the ShootDefaults in the Seed, we don't error and let Kubernetes default it. For shoot with workers, we error in this case. Maybe we can do the same for workerless.

Reply:
OK, I'm fine with keeping `.spec.networking` optional from API PoV to not force people to provide `.spec.networking={}` when they want to rely on defaulting. However, also as said in other comments, we should always default the service CIDR and not rely on on Kubernetes API defaulting.

Reply:
Address in 1af5fb1cdad6e88660545409a0cb3e2de6778279. PTAL.
Now if serviceIP is not given in shoot yaml and also it can't get defaulted because seed doesn't have shoot defaults then it will throw a validation error that serviceIPs are required.